source/robot_model/tests/test_robot_model.cpp

```diff
@@ -108,6 +108,19 @@ TEST_F(RobotModelTest, TestForwardGeometryInvalidFrameName) {
   EXPECT_THROW(franka->forward_geometry(joint_state, "panda_link99"), exceptions::FrameNotFoundException);
 }
 
+TEST_F(RobotModelTest, TestInverseGeometry) {
+  bool success;
+  for (std::size_t config = 0; config < test_configs.size(); ++config) {
+    success=false;
+    state_representation::JointPositions q = franka->inverse_geometry(franka->forward_geometry(test_configs[config],"panda_link8"), success, "panda_link8", 1e-4, 1000);
+    EXPECT_TRUE(success);
+    for (int i = 0; i < 7; ++i) {
+      EXPECT_NEAR(test_configs[config].get_positions()[i], q.get_positions()[i], 1e-4);
```

Comment:
This is not a good test case (and indeed currently fails) because the joint-space solution for any end-effector pose is not unique. Therefore it's very unlikely that you will have exactly the same joint configuration solution as the input configuration.

You should store the expected state from `franka->forward_geometry(test_configs[config],"panda_link8")` and then compare that to the forward geometry of your output `q`.

Reply:
yes. right. Changed.